vendor/go.etcd.io/etcd/client/v3/kubernetes.go

```diff
@@ -0,0 +1,188 @@
+package clientv3
+
+import (
+	"context"
+
+	pb "go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+)
+
+func NewKubernetes(c *Client) Kubernetes {
+	return &kubernetes{kv: RetryKVClient(c)}
+}
+
+type Kubernetes interface {
+	Get(ctx context.Context, key string, opts GetOptions) (KubernetesGetResponse, error)
+	List(ctx context.Context, prefix string, opts ListOptions) (KubernetesListResponse, error)
+	Count(ctx context.Context, prefix string) (int64, error)
+	OptimisticPut(ctx context.Context, key string, value []byte, opts PutOptions) (KubernetesPutResponse, error)
+	OptimisticDelete(ctx context.Context, key string, opts DeleteOptions) (KubernetesDeleteResponse, error)
+}
+
+type GetOptions struct {
+	Revision int64
+}
+
+type ListOptions struct {
+	Revision int64
+	Limit    int64
+	Continue string
+}
+
+type PutOptions struct {
+	ExpectedRevision int64
+	GetOnFailure     bool
+	// Deprecated: Should be replaced with TTL when Kubernetes starts using one lease per object.
+	LeaseID LeaseID
+}
+
+type DeleteOptions struct {
+	ExpectedRevision int64
+	GetOnFailure     bool
+}
+
+type KubernetesGetResponse struct {
+	KV       *mvccpb.KeyValue
+	Revision int64
+}
+
+type KubernetesListResponse struct {
+	KVs      []*mvccpb.KeyValue
+	Count    int64
+	Revision int64
+}
+
+type KubernetesPutResponse struct {
+	KV        *mvccpb.KeyValue
+	Succeeded bool
+	Revision  int64
+}
+
+type KubernetesDeleteResponse struct {
+	KV        *mvccpb.KeyValue
+	Succeeded bool
+	Revision  int64
+}
+
+type kubernetes struct {
+	kv pb.KVClient
+}
+
+func (k kubernetes) Get(ctx context.Context, key string, opts GetOptions) (resp KubernetesGetResponse, err error) {
```

Comment:
Do we need test coverage for this new type?

Reply:
Yes, will add them when we agree on the base design.